__tests__/private.ts

```diff
@@ -547,6 +568,48 @@ describe('POST /p/newUser', () => {
       variation: 1,
     });
   });
+
+  it('should add a new user with language', async () => {
+    const { body } = await request(app.server)
+      .post('/p/newUser')
+      .set('Content-type', 'application/json')
+      .set('authorization', `Service ${process.env.ACCESS_SECRET}`)
+      .send({
+        id: usersFixture[0].id,
+        name: usersFixture[0].name,
+        image: usersFixture[0].image,
+        username: usersFixture[0].username,
+        email: usersFixture[0].email,
+        experienceLevel: 'foo',
+        language: ContentLanguage.English,
+      })
+      .expect(200);
+
+    expect(body).toEqual({ status: 'ok', userId: usersFixture[0].id });
+
+    const users = await con.getRepository(User).find({ order: { id: 'ASC' } });
+    expect(users[0].id).toEqual(usersFixture[0].id);
+    expect(users[0].language).toEqual(ContentLanguage.English);
+  });
+
+  it('should not add a new user with invalid language', async () => {
+    const { body } = await request(app.server)
+      .post('/p/newUser')
+      .set('Content-type', 'application/json')
+      .set('authorization', `Service ${process.env.ACCESS_SECRET}`)
+      .send({
+        id: usersFixture[0].id,
+        name: usersFixture[0].name,
+        image: usersFixture[0].image,
+        username: usersFixture[0].username,
+        email: usersFixture[0].email,
+        experienceLevel: 'foo',
+        language: 'klingon',
+      })
+      .expect(200);
+
+    expect(body).toEqual({ status: 'failed', reason: 'MISSING_FIELDS' });
```

Comment:
We actually need to check maybe with Viktor what happens now if we throw some error to heimdall?
Not really sure it persists back to the end user. (which would mean probably half broken user exists somewhere)

Reply:
It should be the same as when we introduced experienceLevel, so should be no problem?

Reply:
On the other thought, maybe experienceLevel was introduced as required later due to extension, I can just make it optional for now until adoption/experiment is over?

Reply:
It needs to be optional for flows where we subscribe first -> then ask for it.
I think email is that exception if I remember correct.
 (Actually not sure how we fixed that now for expLevel?) CC: @nensidosari

Reply:
Required experienceLevel was introduced here https://github.com/dailydotdev/daily-api/pull/2034, see thread for more info

Reply:
I think we don't need language to be required until experiment concludes either way?

Reply:
Oh I see: `if (data?.username` fixes the email users :) 
Hmm yeah just have it optional I guess for now.

Reply:
Made optional here 58bc9a9ab43f7f34d3074256a418e27825be06b4 but still validates it if provided